src/ca11y/index.js

```diff
@@ -103,23 +229,47 @@ class Ca11y {
     }, 100)
   }
 
+  /**
+   * Click event handler for each day displayed in the calendar. Parses
+   * which day was clicked and delegates to `selectDay`.
+   * @param  { Object } e - the event object
+   * @return { Void }
+   **/
   onDayClick(e) {
     const day = Number(e.target.textContent)
     this.selectDay(day)
   }
 
-  selectDay(day, keepOpen, silent) {
-    if (!silent) {
-      this.ui.input.value = `${zeroPad(this.state.monthDisplay)}/${zeroPad(day)}/${this.state.fullYear}`
-    }
-    this.setState({day})
+  /**
+   * Selects a day, updating internal state and, if `silent` is true, the input's
+   * value as well. If `keepOpen` is not true, the datepicker is hidden afterwards.
+   *
+   * @param  { Number }  day      - 1-indexed day number to select
+   * @param  { Boolean } keepOpen - keep the datepicker open after selecting the day?
+   * @param  { Boolean } silent   - re-render after updates?
+   * @return { Void }
+   **/
+  selectDay(day, keepOpen=false, silent=false) {
+    const { month, fullYear } = this.state
+    const date = new Date(fullYear, month, day)
+
+    this.setDate(date, silent)
+
+    // update the input to reflect new state
+    if (!silent) this.ui.input.value = this.props.format(this.state.date)
```

Comment:
Sort of an architectural question, but should this mutation go here or in a render/update method?